mgradm/shared/templates/postUpgradeScriptTemplate.go

```diff
@@ -11,16 +11,15 @@ import (
 
 const postUpgradeScriptTemplate = `#!/bin/bash
 sed 's/cobbler\.host.*/cobbler\.host = localhost/' -i /etc/rhn/rhn.conf;
-grep uyuni_authentication_endpoint /etc/cobbler/settings.yaml
-if [ $? -eq 1 ]; then
-	echo 'uyuni_authentication_endpoint: "http://localhost"' >> /etc/cobbler/settings.yaml
+if grep uyuni_authentication_endpoint /etc/cobbler/settings.d/zz-uyuni.settings; then
+	echo 'uyuni_authentication_endpoint: "http://localhost"' >> /etc/cobbler/settings.d/zz-uyuni.settings
 else
 	sed 's/uyuni_authentication_endpoint.*/uyuni_authentication_endpoint: http:\/\/localhost/' \
-        -i /etc/cobbler/settings.yaml;
+        -i /etc/cobbler/settings.d/zz-uyuni.settings;
 fi
 
 grep pam_auth_service /etc/rhn/rhn.conf
-if [ $? -eq 1 ]; then
+if grep pam_auth_service /etc/rhn/rhn.conf; then
```

Comment:
Also you can use `-q` option to silence the output.
```suggestion
if grep -q pam_auth_service /etc/rhn/rhn.conf; then
```